presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ImplementFilteredAggregations.java

```diff
@@ -105,15 +117,34 @@ public Result apply(AggregationNode aggregation, Captures captures, Context cont
         return Result.ofPlanNode(
                 new AggregationNode(
                         context.getIdAllocator().getNextId(),
-                        new ProjectNode(
+                        new FilterNode(
                                 context.getIdAllocator().getNextId(),
-                                aggregation.getSource(),
-                                newAssignments.build()),
+                                new ProjectNode(
+                                        context.getIdAllocator().getNextId(),
+                                        aggregation.getSource(),
+                                        newAssignments.build()),
+                                combineDisjunctsWithDefault(filterExpressions.build(), TRUE_LITERAL)),
```

Comment:
let's do not add `FilterNode` when `combineDisjunctsWithDefault(filterExpressions.build(), TRUE_LITERAL)` evaluates to `TRUE` and please add return`ValuesNode` when evaluates to `FALSE`. Make sure you covered this cases with unit tests (checking the query output and plan).

Reply:
As discussed offline, I will add tests only in this PR and let the upper optimiser rules handle this.